core/src/main/java/com/google/errorprone/bugpatterns/UnusedVariable.java

```diff
@@ -223,6 +223,12 @@ public Description matchCompilationUnit(CompilationUnitTree tree, VisitorState s
       if (onlyCheckForReassignments.contains(unusedSymbol) && specs.size() <= 1) {
         continue;
       }
+
+      // Don't complain if the parent of this variable is a Record
+      if(allUsageSites.stream().anyMatch(e -> e.getParentPath().getLeaf().getKind()== Tree.Kind.RECORD)){
+        continue;
+      }
```

Comment:
I'm wondering if this could miss actual unused variables inside record declarations. I'm going to try skipping the synthetic fields for record components with something like the following, but please let me know if you see any issues with that alternative:

```diff
diff --git a/core/src/main/java/com/google/errorprone/bugpatterns/UnusedVariable.java b/core/src/main/java/com/google/errorprone/bugpatterns/UnusedVariable.java
index 3c347ccc9..26a13ced2 100644
--- a/core/src/main/java/com/google/errorprone/bugpatterns/UnusedVariable.java
+++ b/core/src/main/java/com/google/errorprone/bugpatterns/UnusedVariable.java
@@ -85,6 +85,7 @@ import com.sun.source.util.SimpleTreeVisitor;
 import com.sun.source.util.TreePath;
 import com.sun.source.util.TreePathScanner;
 import com.sun.source.util.TreeScanner;
+import com.sun.tools.javac.code.Flags;
 import com.sun.tools.javac.code.Symbol;
 import com.sun.tools.javac.code.Symbol.MethodSymbol;
 import com.sun.tools.javac.code.Symbol.TypeSymbol;
@@ -610,10 +611,15 @@ public final class UnusedVariable extends BugChecker implements CompilationUnitT
           && ASTHelpers.hasDirectAnnotationWithSimpleName(variableTree, "Inject")) {
         return true;
       }
+      if ((symbol.flags() & RECORD_FLAG) == RECORD_FLAG) {
+        return false;
+      }
       return variableTree.getModifiers().getFlags().contains(Modifier.PRIVATE)
           && !SPECIAL_FIELDS.contains(symbol.getSimpleName().toString());
     }
 
+    private static final long RECORD_FLAG = 1L << 61;
+
     /** Returns whether {@code sym} can be removed without updating call sites in other files. */
     private boolean isParameterSubjectToAnalysis(Symbol sym) {
       checkArgument(sym.getKind() == ElementKind.PARAMETER);
```

Reply:
They can't really be unused though right? Since they automatically get a public accessor method.

Reply:
Record components can't be unused, but I wanted to make it clearer that stuff like the following was still handled:

```java
  public record R (Integer foo, Long bar) {
    void f() {
      int x = foo; // unused
    }
  }
```

Reply:
Makes sense. good catch.